### PR TITLE
Folasade/imp 5 supabase client

### PR DESCRIPTION
**Pull request by foloye**
### Description

[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

In this PR, I have initialized the supabase, and created some dummy queries to test if the client is working properly.

### Screenshots

[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

Screenshot 2023-10-08 at 12 58 23 AM

## How to review
To test queries, a user must go to impact fund supabase -> click authentication tab -> disable rls for cases table -> add console.logs displaying data in impact-fund/lib/DummyQueries.tsx -> run app -> click Get Case Data button

(sorry that it's kinda hard to test, didn't want to push console logs!)

[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

First a user should add console.logs to the DummyQueries file, then change function within onPress to test out various queries. This should just be reviewed all at once. Edge cases should mainly be tested out when retool app is up and functioning.

## Next steps

[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

This PR is mainly to test out some queries on the database, however the real testing will come when it is connected to retool and user auth is set up.

### Online sources

[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @stephaniewong2

### CODE REVIEW

File: lib/supabase.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { createClient } from '@supabase/supabase-js';
+import 'react-native-url-polyfill/auto';
+
+const supabaseUrl = 'https://kvfpmjezholwvgdmabhr.supabase.co';
+const supabaseAnonKey =
+  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Imt2ZnBtamV6aG9sd3ZnZG1hYmhyIiwicm9sZSI6ImFub24iLCJpYXQiOjE2OTUwNzAzMDIsImV4cCI6MjAxMDY0NjMwMn0.e6CFmLq4kjNnoGzHDd_hVSUZI6gHaDU-dWu_1Uhd6Mk';
```
Comment (stephaniewong2) [inline lib/supabase.ts]: We don't want to expose the keys. Using the `react-native-dotenv` package allows us to do this

File: lib/supabase.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { createClient } from '@supabase/supabase-js';
+import 'react-native-url-polyfill/auto';
```
Comment (stephaniewong2) [inline lib/supabase.ts:2]: What is this used for?

File: src/app/index.tsx [hunk 3]
```diff
@@ -1,16 +1,38 @@
...
+import { useEffect, useState } from 'react';
 import { Text, View } from 'react-native';
 import { TouchableOpacity } from 'react-native-gesture-handler';
+import 'react-native-url-polyfill/auto';
+import DummyQueries from '../../lib/DummyQueries';
+import { supabase } from '../../lib/supabase';
 import styles from './styles';
 
 function StartScreen() {
+  const [session, setSession] = useState<Session | null>(null);
```
Comment (stephaniewong2) [inline src/app/index.tsx]: eslint is complaining about `eslint[@typescript-eslint/no-shadow](https://typescript-eslint.io/rules/no-shadow)` and we can resolve this by renaming the state variable `session`.

Comment (stephaniewong2) [reply src/app/index.tsx]: Also eslint will complain if we do not use this variable so for now, we can either remove it or get the session user in order to use the variable.

File: lib/DummyQueries.tsx
Comment (stephaniewong2) [inline lib/DummyQueries.tsx:1]: eslint is complaining about unused functions. Could you also add buttons to run all the other queries? Thank you! Besides that, looks good to me!

File: src/app/index.tsx [hunk 4]
```diff
@@ -1,16 +1,38 @@
...
+  }, []);
+
   return (
     <View style={styles.container}>
       <Link href="/Welcome" asChild>
         <TouchableOpacity style={styles.button}>
           <Text>Welcome Screen</Text>
         </TouchableOpacity>
       </Link>
+      <Link href="/SignUp" asChild>
```
Comment (stephaniewong2) [inline src/app/index.tsx]: Remove `SignUp` link. This is in the auth flow from the Welcome page!

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]

File: src/app/index.tsx [hunk 5]
```diff
@@ -1,16 +1,38 @@
...
+
+  useEffect(() => {
+    supabase.auth.getSession().then(({ data: { session } }) => {
+      setSession(session);
+    });
+
+    supabase.auth.onAuthStateChange((_event, session) => {
+      setSession(session);
+    });
+  }, []);
```
Comment (stephaniewong2) [inline src/app/index.tsx]: We should add a clean-up function on return (unsubscribe from the session) using something like `authListener.subscription.unsubscribe();`

File: src/app/index.tsx (hunk 3 above)
Comment (stephaniewong2) [reply src/app/index.tsx]: Let's keep it because we will use it shortly!

File: lib/DummyQueries.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      case_site: 'berkeley.edu',
+      opt_out_link: 'berkeley.edu',
+    };
+    const { error } = await supabase.from('Cases').insert(dummyCase);
+    queryHolder = error;
+  }
+
+  return (
+    <View>
+      <Button title="Get Case Data" onPress={() => allCases()} />
```
Comment (stephaniewong2) [inline lib/DummyQueries.tsx]: Very minor but let's use `TouchableOpacity` instead of `Button` and make texts specific to queries! Can you also add a little bit of styling or copy over from the other buttons above? Besides that, everything looks good!

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]

File: lib/DummyQueries.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
+import { View } from 'react-native';
+import { Text } from 'react-native-elements';
+import { TouchableOpacity } from 'react-native-gesture-handler';
```
Comment (stephaniewong2) [inline lib/DummyQueries.tsx]: Let's use `TouchableOpacity` from `react-native` instead!

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]

**Review by stephaniewong2 — APPROVED** [review]
